src/utils.cxx

```diff
@@ -374,11 +443,11 @@ std::string stem(const std::string &file)
 
 std::string basename(const std::string &file)
```

Comment:
Can you add a docstring that explains:

`/x/y` -> `y`
`/x/y/` -> ``

(this behavior is fine, it matches Python's `os.path.basename`, but it's worth stating explicitly)

Reply:
Will do!